Add namespace support to service discovery

pgressa opened the pull request.

pgressa commented:
This PR is missing the configuration of k8s env, that will be discussed with @alvarosanchez tmrw.

pgressa commented:
@alvarosanchez please check

alvarosanchez commented on setup-test-namespace.sh:
```diff
@@ -0,0 +1,166 @@
+kubectl label secret another-secret app=game
+
+# Create and expose deployment - example service
+cat <<EOF | kubectl apply -f -
```
Could these YAML snippets be stored on their own YAML files?

pgressa replied:
ok

alvarosanchez reviewed (approved):
Left one comment. Other than that, this is good to go.

pgressa commented:
@alvarosanchez addressed your comment, but seems like the test is flaky a bit, checkout https://github.com/micronaut-projects/micronaut-kubernetes/pull/191/checks?check_run_id=1248388611#step:7:101 , in other checks from that run it passes. When I checked the previous ones, this exact test fails there as well.

alvarosanchez commented on kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/client/v1/KubernetesServiceConfiguration.java:
```diff
@@ -0,0 +1,97 @@
+import java.util.Optional;
+
+/**
+ * Discovery configuration for Kubernetes service.
```
Needs `@since` tag
